packages/financial-templates-lib/test/price-feed/DefiPulseTotalPriceFeed.js

```diff
@@ -0,0 +1,91 @@
+const { DefiPulseTotalPriceFeed } = require("../../src/price-feed/DefiPulseTotalPriceFeed");
+const { NetworkerMock } = require("../../src/price-feed/NetworkerMock");
+const winston = require("winston");
+var Web3 = require("web3");
+var web3 = new Web3("http://localhost:8545");
+
+contract("DefiPulseTotalPriceFeed.js", function() {
+  let defiPulseTotalPriceFeed;
+  let mockTime = 1611583300;
+  let networker;
+
+  const apiKey = "test-api-key";
+
+  const lookback = 3600 * 24 * 7;
+
+  const getTime = () => mockTime;
+  const minTimeBetweenUpdates = 600; // every 10 minutes
+
+  // Fake data to inject.
+  const validResponses = [
+    [
+      { timestamp: "1611572400", tvlUSD: 24780000000 },
+      { timestamp: "1611576000", tvlUSD: 23500000000 },
+      { timestamp: "1611579600", tvlUSD: 22250000000 },
+      { timestamp: "1611583200", tvlUSD: 25100000001 }
+    ]
+  ];
+
+  beforeEach(async function() {
+    networker = new NetworkerMock();
+    const dummyLogger = winston.createLogger({
+      level: "info",
+      transports: [new winston.transports.Console()]
+    });
+
+    defiPulseTotalPriceFeed = new DefiPulseTotalPriceFeed(
+      dummyLogger,
+      web3,
+      apiKey,
+      lookback,
+      networker,
+      getTime,
+      minTimeBetweenUpdates,
+      6 // Add arbitrary decimal conversion and prove this works.
+    );
+  });
+
+  it("No update", async function() {
+    assert.equal(defiPulseTotalPriceFeed.getCurrentPrice(), undefined);
+    assert.isTrue(await defiPulseTotalPriceFeed.getHistoricalPrice(1000).catch(() => true));
+    assert.equal(defiPulseTotalPriceFeed.getLastUpdateTime(), undefined);
+  });
+
+  it("Check decimals", async function() {
+    assert.equal(defiPulseTotalPriceFeed.getPriceFeedDecimals(), 6);
+  });
+
+  it("Basic historical price", async function() {
+    // Inject data.
+    networker.getJsonReturns = [...validResponses];
+
+    await defiPulseTotalPriceFeed.update();
+
+    // Before period 1 should return null.
+    assert.isTrue(await defiPulseTotalPriceFeed.getHistoricalPrice(1611572399).catch(() => true));
+    assert.equal((await defiPulseTotalPriceFeed.getHistoricalPrice(1611575900)).toString(), web3.utils.toWei("24.78"));
+    assert.equal((await defiPulseTotalPriceFeed.getHistoricalPrice(1611579500)).toString(), web3.utils.toWei("23.50"));
+    assert.equal((await defiPulseTotalPriceFeed.getHistoricalPrice(1611579500)).toString(), web3.utils.toWei("23.50"));
+    assert.equal((await defiPulseTotalPriceFeed.getHistoricalPrice(1611579601)).toString(), web3.utils.toWei("22.25"));
```

Comment:
Since the decimals are specified as 6, shouldn't these be coming out as 24.78 * 10^6? Right now, it looks like they're coming out as 24.78 * 10^18. See my comment above for how to make the price feed to do what you expect here.
```suggestion
    assert.equal((await defiPulseTotalPriceFeed.getHistoricalPrice(1611575900)).toString(), parseFixed("24.78", decimals).toString());
    assert.equal((await defiPulseTotalPriceFeed.getHistoricalPrice(1611579500)).toString(), parseFixed("23.50", decimals).toString());
    assert.equal((await defiPulseTotalPriceFeed.getHistoricalPrice(1611579500)).toString(), parseFixed("23.50", decimals).toString());
    assert.equal((await defiPulseTotalPriceFeed.getHistoricalPrice(1611579601)).toString(), parseFixed("22.25", decimals).toString());
```